graphql/src/test/java/com/linecorp/armeria/server/graphql/GraphqlServiceMultipartTest.java

```diff
@@ -0,0 +1,149 @@
+import com.linecorp.armeria.common.multipart.Multipart;
+import com.linecorp.armeria.common.multipart.MultipartFile;
+import com.linecorp.armeria.server.ServerBuilder;
+import com.linecorp.armeria.testing.junit5.server.ServerExtension;
+
+import graphql.schema.DataFetcher;
+import graphql.schema.idl.errors.SchemaProblem;
+
+class GraphqlServiceMultipartTest {
+    @RegisterExtension
+    static ServerExtension server = new ServerExtension() {
+        @Override
+        protected void configure(ServerBuilder sb) throws Exception {
+            final File graphqlSchemaFile =
+                    new File(getClass().getResource("/multipart.graphqls").toURI());
+            final GraphqlService service =
+                    GraphqlService.builder()
+                                  .schemaFile(graphqlSchemaFile)
+                                  .runtimeWiring(c -> {
+                                      c.scalar(MoreScalars.multipartFile());
+                                      final DataFetcher<String> fileUpload = fileUploadFetcher();
+                                      c.type("Mutation",
+                                             typeWiring -> typeWiring.dataFetcher("fileUpload", fileUpload));
+                                      final DataFetcher<List<String>> fileUploads = fileUploadsFetcher();
+                                      c.type("Mutation",
+                                             typeWiring -> typeWiring.dataFetcher("fileUploads", fileUploads));
+                                  })
+                                  .build();
+            sb.service("/graphql", service);
+        }
+    };
+
+    private static DataFetcher<String> fileUploadFetcher() {
+        return environment -> CompletableFuture.supplyAsync(() -> {
+            try {
+                final MultipartFile multipartFile = environment.getArgument("file");
+                return Files.asCharSource(multipartFile.file(), StandardCharsets.UTF_8).read();
+            } catch (IOException e) {
+                throw new UncheckedIOException(e);
+            }
+        }, GraphqlServiceContexts.get(environment).blockingTaskExecutor()).join();
+    }
+
+    private static DataFetcher<List<String>> fileUploadsFetcher() {
+        return environment -> CompletableFuture.supplyAsync(() -> {
+            final List<MultipartFile> multipartFiles = environment.getArgument("files");
+            return multipartFiles.stream()
+                        .map(it -> {
+                            try {
+                                return Files.asCharSource(it.file(), StandardCharsets.UTF_8).read();
+                            } catch (IOException e) {
+                                throw new UncheckedIOException(e);
+                            }
+                        })
+                        .collect(toImmutableList());
+        }, GraphqlServiceContexts.get(environment).blockingTaskExecutor()).join();
```

Comment:
Ditto. remove `join()` and return `DataFetcher<CompletableFuture<List<String>>>`?